src/Raven.Client/Documents/AI/DatabaseAiAgents.cs

```diff
@@ -0,0 +1,186 @@
+    /// <param name="configuration">The configuration to assign to the agent.</param>
+    /// <returns>The result of the creation or update operation.</returns>
+    public async Task<AiAgentConfigurationResult> CreateAgentAsync<TSchema>(string agentName, AiAgentConfiguration configuration, CancellationToken token = default) where TSchema : new()
+    {
+        return await _store.Maintenance.ForDatabase(_databaseName).SendAsync(new AddOrUpdateAiAgentOperation<TSchema>(agentName, configuration), token).ConfigureAwait(false);
+    }
+
+    /// <summary>
+    /// Creates or updates an AI agent configuration (with the given schema) on the database.
+    /// </summary>
+    /// <typeparam name="TSchema">The schema type the AI agent should use.</typeparam>
+    /// <param name="agentName">The unique name of the AI agent for create an agent, or name of an existed agent for update an agent.</param>
+    /// <param name="configuration">The configuration to assign to the agent.</param>
+    /// <returns>The result of the creation or update operation.</returns>
+    public AiAgentConfigurationResult CreateAgent<TSchema>(string agentName, AiAgentConfiguration configuration) where TSchema : new()
+    {
+        return AsyncHelpers.RunSync(() => CreateAgentAsync<TSchema>(agentName, configuration));
+    }
+
+    /// <summary>
+    /// Asynchronously starts a new chat with an AI agent using a fluent parameter builder.
+    /// </summary>
+    /// <typeparam name="TSchema">The schema type for the chat.</typeparam>
+    /// <param name="agentName">The name of the AI agent to chat with.</param>
+    /// <param name="prompt">The initial user prompt.</param>
+    /// <param name="func">A builder function to define RAG parameters.</param>
+    /// <returns>The result of the chat.</returns>
+    public Task<ChatResult<TSchema>> StartChatAsync<TSchema>(string agentName, string prompt, Func<AiAgentParametersBuilder, AiAgentParametersBuilder> func, CancellationToken token = default) where TSchema : new()
+    {
+        var builder = func.Invoke(new AiAgentParametersBuilder());
+        var parameters = builder.GetParameters();
+        return StartChatAsync<TSchema>(agentName, prompt, parameters, token);
+    }
+
+    /// <summary>
+    /// Starts a new chat with an AI agent using a fluent parameter builder.
+    /// </summary>
+    /// <typeparam name="TSchema">The schema type for the chat.</typeparam>
+    /// <param name="agentName">The name of the AI agent to chat with.</param>
+    /// <param name="prompt">The initial user prompt.</param>
+    /// <param name="func">A builder function to define RAG parameters.</param>
+    /// <returns>The result of the chat.</returns>
+    public ChatResult<TSchema> StartChat<TSchema>(string agentName, string prompt, Func<AiAgentParametersBuilder, AiAgentParametersBuilder> func) where TSchema : new()
+    {
+        var builder = func.Invoke(new AiAgentParametersBuilder());
```

Comment:
Why use this method instead of allowing you to pass `Dictionary<string, object>` ?